tests/python/pants_test/reporting/test_reporting_remote_integration.py

```diff
@@ -0,0 +1,87 @@
+from pants.util.contextutil import http_server, temporary_dir
+from pants.util.reporting_util import zipkin_handler
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest
+
+
+class TestReportingRemoteIntegration(PantsRunIntegrationTest, unittest.TestCase):
+  @contextmanager
+  def run_mock_server(self, bin_name: str, arguments: List[str]):
+    """
+    Build and run a Rust mock server binary.
+    Mainly used for remoting integration tests.
+    It assumes that the server will print a line like `localhost:99999` to indicate its port.
+    :param bin_name: The name of the binary that you want to run. It must be in the main engine's workspace.
+    :param arguments: Passthrough arguments to pass to the binary.
+    """
+    args = ["./build-support/bin/native/cargo.sh", "run", "--manifest-path", "src/rust/engine/Cargo.toml", "--release", "--bin", bin_name]
+    if len(arguments) > 0:
+      args = args + ["--"] + arguments
+    try:
+      process = subprocess.Popen(args, stdin = subprocess.PIPE, stdout = subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines = True)
+      port = None
+      while port is None:
+        match = re.search("localhost:([0-9]+)", process.stdout.readline())
+        if match is not None:
+          port = int(match.group(1))
+      yield port
+    finally:
+      # Terminate the server process by sending it an end of line character
+      process.communicate("\n")
+
+  @contextmanager
+  def run_cas_server(self):
+    with self.run_mock_server("local_cas", []) as port:
+      yield port
+
+  @contextmanager
+  def run_execution_server(self, request_hash, request_size):
+    with self.run_mock_server("local_execution_server", ["--request_hash", request_hash, "--request_size", request_size]) as port:
+      yield port
+
+  def test_zipkin_reporter_for_remote_execution_with_v2_engine(self):
+    # Hardcode the hash and size of the request we know our call to cloc will trigger
+    # If this changes for some reason, the local mock execution server will print out
+    # the digest of the request that was expected and this can be manually updated.
+    request_hash = "b22d7feb653a82f63961c9335922c523a42d588139f493ad78442d29c8e31928"
+    request_size = "142"
+    ZipkinHandler = zipkin_handler()
+    with self.run_cas_server() as cas_port, self.run_execution_server(request_hash, request_size) as execution_port, http_server(ZipkinHandler) as zipkin_port, temporary_dir() as store_dir:
+      endpoint = "http://localhost:{}".format(zipkin_port)
+      command = [
+        'cloc',
+        'src/python/pants:version',
+        '--reporting-zipkin-trace-v2',
+        '--reporting-zipkin-endpoint={}'.format(endpoint),
+        '--remote-execution',
+        '--remote-execution-server=localhost:{}'.format(execution_port),
+        '--remote-store-server=localhost:{}'.format(cas_port),
+        '--cache-cloc-ignore',
+        '--local-store-dir={}'.format(store_dir)
```

Comment:
F strings encouraged throughout this whole PR:
```suggestion
        f'--local-store-dir={store_dir}'
```